x-pack/solutions/security/plugins/security_solution/public/flyout/document_details/right/components/highlighted_fields_modal.tsx

```diff
@@ -0,0 +1,270 @@
+    [fieldOptions]
+  );
+
+  const { form } = useForm({
+    defaultValue: { customHighlightedFields: [] },
+    schema,
+  });
+
+  const onCancel = useCallback(() => {
+    setIsModalVisible(false);
+  }, [setIsModalVisible]);
+
+  const onSubmit = useCallback(async () => {
+    setIsEditLoading(true);
+
+    const updatedRule = await updateRule({
+      ...rule,
+      id: undefined,
+      investigation_fields:
+        selectedOptions.length > 0
+          ? { field_names: selectedOptions.map((option) => option.label) }
+          : undefined,
+    } as RuleUpdateProps);
+
+    addSuccess(SUCCESSFULLY_SAVED_RULE(updatedRule?.name ?? 'rule'));
+    setIsEditLoading(false);
+    setIsModalVisible(false);
+  }, [updateRule, addSuccess, rule, setIsModalVisible, setIsEditLoading, selectedOptions]);
+
+  return (
+    <EuiModal
+      css={{ width: '600px' }}
+      data-test-subj={HIGHLIGHTED_FIELDS_MODAL_TEST_ID}
+      id={modalTitleId}
+      onClose={onCancel}
+    >
+      <EuiModalHeader data-test-subj={HIGHLIGHTED_FIELDS_MODAL_TITLE_TEST_ID}>
+        <EuiModalHeaderTitle id={modalTitleId} size="s">
+          <FormattedMessage
+            id="xpack.securitySolution.flyout.right.investigation.highlightedFields.modalTitle"
+            defaultMessage="Edit highlighted fields"
+          />
+        </EuiModalHeaderTitle>
+      </EuiModalHeader>
+      <EuiModalBody>
+        <EuiText size="s" data-test-subj={HIGHLIGHTED_FIELDS_MODAL_DESCRIPTION_TEST_ID}>
+          <FormattedMessage
+            id="xpack.securitySolution.flyout.right.investigation.highlightedFields.modalDescription"
+            defaultMessage="Changes made here will be applied to the {ruleName} rule. Any custom fields you add will be displayed in all alerts generated by this rule."
+            values={{ ruleName: <b>{rule.name}</b> }}
+          />
+        </EuiText>
+        <EuiSpacer size="m" />
+        <EuiText
+          size="xs"
+          css={css`
+            font-weight: ${euiTheme.font.weight.semiBold};
+          `}
+        >
+          <FormattedMessage
+            id="xpack.securitySolution.flyout.right.investigation.highlightedFields.modalDefaultFieldsTitle"
+            defaultMessage="Default"
+          />
+        </EuiText>
+        <EuiSpacer size="s" />
+        <EuiFlexGroup
+          data-test-subj={HIGHLIGHTED_FIELDS_MODAL_DEFAULT_FIELDS_TEST_ID}
+          gutterSize="xs"
+          justifyContent="flexStart"
+          wrap
+        >
+          {defaultFieldsArray.map((field: string) => (
+            <EuiFlexItem key={field} grow={false}>
+              <EuiBadge color="hollow">{field}</EuiBadge>
+            </EuiFlexItem>
+          ))}
+        </EuiFlexGroup>
+        <EuiSpacer size="l" />
+        <Form form={form}>
+          <CommonUseField
+            path="customHighlightedFields"
+            config={{ ...schema.customHighlightedFields, label: ADD_CUSTOM_FIELD_LABEL }}
```

Comment:
This could be extracted into a constant outside of the component to avoid creating a new object every time
```typescript
{ ...schema.customHighlightedFields, label: ADD_CUSTOM_FIELD_LABEL }
```